### ISSUE TITLE
Add region landmarks to Search Results page

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13517]
**Describe the bug**

- Screen Reader users must be able to navigate by landmarks, currently, there aren't many nor labelled, therefore relevant sections on the page should be identified with meaningful landmarks names.

**Tell us the version of Spartacus**
- Library version: 4.0.0.

**To Reproduce**
1. Open any Screen Reader (VO, NVDA, JAWS).
2. Bring up the Rotor or Element List Menu.
3. Choose Landmarks view.
4. See absence of relevant landmarks.

**Expected behavior**

- Use only predefined landmarks https://wiki.wdf.sap.corp/wiki/display/spar/Landmarks
- `<cx-product-facet-navigation role="region" aria-label="Filter Results by Facets">`
- `<cx-product-list role="region" aria-label="Product Results List">`

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.
<img width="1440" alt="Screen Shot 2021-12-13 at 11 52 28 AM" src="https://user-images.githubusercontent.com/45409515/145866170-142b28d4-ba60-4db3-8d3b-698735b1d97c.png">
<img width="1440" alt="Screen Shot 2021-12-13 at 11 51 48 AM" src="https://user-images.githubusercontent.com/45409515/145866181-45c09dc2-0ab3-4c15-bbbb-62d341af1b1f.png">
